elastic/kibana
# [Upgrade Assistant] Add upgrade system indices section

### Summary

This PR adds the system indices migration step into the Overview page from Upgrade Assistant. The system indices are grouped by feature and only the status per feature is shown to the user in the flyout since we don't need to bother them with implementation details. ES documentation can be found [here](https://www.elastic.co/guide/en/elasticsearch//reference/master/migration-api-feature-upgrade.html).

Notes: 
* At this time the ES API returns stubs, so clicking the `begin upgrade` won't do anything.
* There's also an [open question](https://www.elastic.co/guide/en/elasticsearch//reference/master/migration-api-feature-upgrade.html#feature-upgrade-api-prereqs) in regards of permissions that might be needed to use this feature but if needed it can be addressed in a subsequent PR.
* Still waiting to get confirmation on how upgrade errors for certain indices will be returned to us to show in the UI.

**How to test**
1. Start up elasticsearch, `yarn es snapshot --license=trial`
3. Start up Kibana, `yarn start` and navigate to UA
4. Verify that clicking `Begin upgrading system indices` starts the migration process
5. Verify that clicking `View status` shows the flyout with the features and their status
6. Verify that we poll for upgrade status only when the migration is `IN_PROGRESS`
7. Verify that the step is marked as completed when upgrade status is `NO_UPGRADE_NEEDED`

At the time of writing ES only returns a stub for both starting the migration and getting the status. If you want to see in more details other aspects of the UI, you can return [a mocked response](https://github.com/elastic/kibana/pull/110593/files#diff-afe3c83820f0da59884f7e9f415547a40c0cde9670da21a74da661de4afd2586R10) from the [get status route](https://github.com/elastic/kibana/pull/110593/files#diff-bb140df9a1ba7bee6d6293fd05a2378aabcba18d002d13794e78d361e8e1517bR88).


**Screenshots**

<details>
<summary>Overview of changes</summary>

// Has to upgrade
![Screenshot 2021-09-02 at 10 33 54](https://user-images.githubusercontent.com/1191206/131811417-8306d8dd-5ecb-4b66-8e1c-2a470e3144b2.png)


// Is upgrading
![Screenshot 2021-09-02 at 10 34 41](https://user-images.githubusercontent.com/1191206/131811471-4be853db-1c98-4f0b-b3fc-d558401ad8e4.png)


// Upgrade in progress
![Screenshot 2021-09-02 at 10 32 37](https://user-images.githubusercontent.com/1191206/131811503-889ef5b8-563d-426e-a807-3866343cdbee.png)


// Loading error state
![Screenshot 2021-09-03 at 10 18 44](https://user-images.githubusercontent.com/1191206/131974061-a1a277d2-9733-4c78-9e28-eca0c1938cc3.png)


// Error starting upgrade
![Screenshot 2021-09-03 at 10 22 36](https://user-images.githubusercontent.com/1191206/131974526-b15de1a8-8db7-4ea6-ae58-c64e7f1ab39c.png)


// Flyout with indices statuses
![Screenshot 2021-10-05 at 15 05 51](https://user-images.githubusercontent.com/1191206/136028996-d0f629a7-bc17-48f8-be3c-8ffb71a8d482.png)

</details>
